marathon/src/main/scala/io/buoyant/marathon/v2/AppIdNamer.scala

```diff
@@ -42,20 +42,29 @@ class AppIdNamer(
       // each time the map of all Apps updates, find the
       // shortest-matching part of `path` that exists as an App ID.
       val possibleIds = (1 to lowercasePath.size).map(lowercasePath.take(_))
-      appsActivity.map { apps =>
+      appsActivity.flatMap { apps =>
         Trace.recordBinary("marathon.path", path.show)
         val found = possibleIds.collectFirst {
-          case app if apps(app) =>
+          case app if apps(app) => {
             Trace.recordBinary("marathon.appId", app.show)
             val residual = path.drop(app.size)
             val id = prefix ++ app
             val addr = getAndMonitorAddr(app)
-            Name.Bound(addr, id, residual)
+
+            val observation = addr.map {
+              case Addr.Neg => Activity.Ok(NameTree.Neg)
+              case Addr.Pending => Activity.Ok(NameTree.Leaf(Name.Bound(addr, id, residual)))
```

Comment:
I think this should be Activity.Pending

Reply:
As `Activity.Pending`, there are quite a few failures
```
[info] - Namer handles looking up /app/id *** FAILED ***              
[info]   com.twitter.util.Activity$Pending$@294151d6 did not equal Ok(Leaf(Path(io.l5d.marathon,service,name))) (AppIdNamerTest.scala:68)    
[info] - Namer handles looking up /app/id case-insensitive *** FAILED ***                                                                    
[info]   com.twitter.util.Activity$Pending$@294151d6 did not equal Ok(Leaf(Path(io.l5d.marathon,service,name))) (AppIdNamerTest.scala:80)    
[info] - Namer updates when blocking call from getAppIds returns *** FAILED ***                                                              
[info]   com.twitter.util.Activity$Pending$@294151d6 did not equal Ok(Leaf(Path(io.l5d.marathon,servicename))) (AppIdNamerTest.scala:93)                
[info] - Addrs update when blocking call for getAddrs returns *** FAILED ***                                                                 
[info]   com.twitter.util.Activity$Pending$@294151d6 is not a NameTree.Leaf[Name.Bound] (AppIdNamerTest.scala:140)                           
[info] - Namer recovers if marathon api fails initially *** FAILED ***                                                                       
[info]   com.twitter.util.Activity$Pending$@294151d6 did not equal Ok(Leaf(Path(io.l5d.marathon,foo,bar))) (AppIdNamerTest.scala:171)        
[info] - Namer returns a cached address when the marathon api goes down *** FAILED ***                                                       
[info]   com.twitter.util.Activity$Pending$@294151d6 did not equal Ok(Leaf(Path(io.l5d.marathon,foo,bar))) (AppIdNamerTest.scala:192)        
[info] - Namer returns failure if getAddrs() fails due to marathon api being down *** FAILED ***                                             
[info]   Failed(com.twitter.finagle.ChannelWriteException) is not a NameTree.Leaf[Name] (AppIdNamerTest.scala:111)                           
[info] - Namer recovers if getAddrs() fails initially *** FAILED ***  
[info]   Failed(com.twitter.finagle.ChannelWriteException) is not a NameTree.Leaf[Name] (AppIdNamerTest.scala:111)
```

As `Activity.Ok(NameTree.Leaf(Name.Bound(...)))` there are still failures, but fewer
```
[info] - Namer returns failure if getAddrs() fails due to marathon api being down *** FAILED ***                                             
[info]   Failed(com.twitter.finagle.ChannelWriteException) is not a NameTree.Leaf[Name] (AppIdNamerTest.scala:111)                           
[info] - Namer recovers if getAddrs() fails initially *** FAILED ***  
[info]   Failed(com.twitter.finagle.ChannelWriteException) is not a NameTree.Leaf[Name] (AppIdNamerTest.scala:111)
```

Still investigating why that is. It may require changing the tests. Afterall, previously all `found` appIds would be bound regardless of the `addr` value